src/main/java/com/faforever/client/tournament/game/TournamentGameService.java

```diff
@@ -0,0 +1,122 @@
+  private static final int SAFETY_CLOSE_NOTIFICATION_SECONDS = 200;
+  private final FafServerAccessor fafServerAccessor;
+  private final I18n i18n;
+  private final Timer timer = new Timer(true);
+  private final UiService uiService;
+  private final GameService gameService;
+  private final NotificationService notificationService;
+  private ImmediateNotification notification;
+  private CompletableFuture<Void> matchFuture;
+
+
+  @Override
+  public void afterPropertiesSet() throws Exception {
+    fafServerAccessor.addEventListener(IsReadyRequest.class, this::onReadRequest);
+    fafServerAccessor.addEventListener(MatchmakerMatchCancelledResponse.class, this::onMatchCanceled);
+    fafServerAccessor.addEventListener(GameLaunchResponse.class, this::onGameLaunch);
+  }
+
+  private void onGameLaunch(GameLaunchResponse gameLaunchResponse) {
+    dismissNotification();
+  }
+
+  private void onMatchCanceled(MatchmakerMatchCancelledResponse matchmakerMatchCancelledResponse) {
+    dismissNotification();
+    cancelMatch();
+    notificationService.addImmediateInfoNotification(
+        "isReady.matchCanceled"
+    );
+  }
+
+  private void onReadRequest(IsReadyRequest isReadyRequest) {
+    log.info("Tournament game is ready, asking user.");
+    if (notification != null) {
+      log.warn("Tournament ready request ignored because tournament is already in progress.");
+      respondToReadyRequest(isReadyRequest.getRequestId(), isReadyRequest);
+      return;
+    }
+    uiService.bringMainStageToFront();
+    final var controller = initializeIsReadyController(isReadyRequest);
+    notification =
+        new ImmediateNotification(i18n.get("isReady.title"), i18n.get("isReady.message", isReadyRequest.getGameName()),
+            Severity.INFO, null, List.of(), controller.getRoot(), false);
+    notificationService.addNotification(notification);
+    safetyCloseMechanism();
+  }
+
+  private void safetyCloseMechanism() {
+    final var currentNotification = notification;
+    timer.schedule(new TimerTask() {
+      @Override
+      public void run() {
+        if (notification != currentNotification) {
+          return;
+        }
+        JavaFxUtil.runLater(() -> dismissNotification());
+      }
+    }, SAFETY_CLOSE_NOTIFICATION_SECONDS * 1000);
+  }
+
+  private void dismissNotification() {
+    if (notification == null) {
+      return;
+    }
+    notification.dismiss();
+    notification = null;
+  }
+
+  @NotNull
+  private IsReadyForGameController initializeIsReadyController(IsReadyRequest isReadyRequest) {
+    IsReadyForGameController controller = uiService.loadFxml("theme/tournaments/is_ready_for_game.fxml");
+    controller.setTimeout(isReadyRequest.getResponseTimeSeconds());
+    controller.setReadyCallback(() -> respondToReadyRequest(isReadyRequest.getRequestId(), isReadyRequest));
+    controller.setDismissCallBack(() -> JavaFxUtil.runLater(this::dismissNotification));
```

Comment:
Might be better to just do the JavaFxUtil wrapping in dismissNotification, to make it more error proof